torch/csrc/autograd/init.cpp

```diff
@@ -1258,7 +1258,37 @@ PyObject* THPModule_increment_version(PyObject* _unused, PyObject* tensor) {
   HANDLE_TH_ERRORS
   TORCH_CHECK(
       THPVariable_Check(tensor), "increment_version expect a Tensor as input");
-  torch::autograd::increment_version((THPVariable_Unpack(tensor)));
+  auto t = THPVariable_Unpack(tensor);
+  torch::autograd::increment_version(std::move(t));
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+
+PyObject* THPModule_increment_versions(
+    PyObject* _unused,
+    PyObject* tensor_or_list) {
+  HANDLE_TH_ERRORS
+  if (PyList_Check(tensor_or_list)) {
```

Comment:
You most likely want to use sequence here https://docs.python.org/3/c-api/sequence.html so you don't have to do 2 passes to create the input in your AOTDispatch code above and can just pass the generator.

Reply:
good point, will try it

Reply:
noob q, should I be using an iterator instead of a sequence? https://docs.python.org/3/c-api/iter.html

I see that sequence exposes a `PySequence_GetItem` that maps to `seq[i]`, although I would imagine that if I'm passing in a generator then I would just want to be looping through the entire generator without using random access.

(e.g. in case, `gen = (args[i] for i in mutated_indices)`